packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -406,69 +406,90 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                 { eventName: "ObtainSnapshot" },
                 async (event: PerformanceEvent) => {
                     const props = {};
-                    let cachedSnapshot: ISnapshotContents | undefined;
+                    let retrievedSnapshot: ISnapshotContents | undefined;
+                    // Here's the logic to grab the persistent cache snapshot implemented by the host
+                    // Epoch tracker is responsible for communicating with the persistent cache, handling epochs and cache versions
                     const cachedSnapshotP: Promise<ISnapshotContents | undefined> =
                         this.epochTracker.get(createCacheSnapshotKey(this.odspResolvedUrl))
-                            .then((snapshotCachedEntry: ISnapshotCachedEntry) => {
+                            .then(async (snapshotCachedEntry: ISnapshotCachedEntry) => {
                                 if (snapshotCachedEntry !== undefined) {
                                     // If the cached entry does not contain the entry time, then assign it a default of 30 days old.
-                                    // eslint-disable-next-line @typescript-eslint/dot-notation
-                                    props["cacheEntryAge"] = Date.now() - (snapshotCachedEntry.cacheEntryTime ??
+                                    const age = Date.now() - (snapshotCachedEntry.cacheEntryTime ??
                                         (Date.now() - 30 * 24 * 60 * 60 * 1000));
+
+                                    // Record the cache age
+                                    // eslint-disable-next-line @typescript-eslint/dot-notation
+                                    props["cacheEntryAge"] = age;
+
+                                    // TODO: Refactor this logic (maybe move to a separate function)
+                                    // TODO: Currently there is a max of two days. Potentially remove this logic once proactive expiry exists.
+                                    // Set the max age to 2 days, a network call to retrieve the snapshot will be made if undefined is returned.
+                                    const twoDays = 2 * 24 * 60 * 60 * 1000;
+                                    const maxCacheAge = twoDays;
+                                    if(age > maxCacheAge) {
+                                        await this.epochTracker.removeEntries();
+                                        this.logger.sendTelemetryEvent({ eventName: "odspVersionsCacheExpired", duration: age, maxCacheAge });
+                                        return undefined;
+                                    }
                                 }
+
                                 return snapshotCachedEntry;
                         });
 
+                    // Setup the network snapshot call
+                    const networkSnapshotP = this.fetchSnapshot(hostSnapshotOptions);
```

Comment:
You are changing behavior by moving it here. Before, ELSE branch below would do network call ONLY if cached retrieval failed, but now it does all the time.

Reply:
I believe (at least for summarizer client) we do not want to make network call if we have cached entry. This is mostly about COGS, i.e. reducing impact on service.

That said, contract clearly states the behavior:

    /**
     * If set to true, tells driver to concurrently fetch snapshot from storage (SPO) and cache
     * Container loads from whatever comes first in such case.
     * Snapshot fetched from storage is pushed to cache in either case.
     * If set to false, driver will first consult with cache. Only on cache miss (cache does not
     * return snapshot), driver will fetch snapshot from storage (and push it to cache), otherwise
     * it will load from cache and not reach out to storage.
     * Passing true results in faster loads and keeping cache more current, but it increases bandwidth consumption.
     */
    concurrentSnapshotFetch?: boolean;